### PR TITLE
Item Materials

### LINKED ISSUE #96: ArmorMaterial identifier is not mod dependent
I just noticed that when registering new armor materials, their name is not an `Identifier`.
This leads to the strange situation that when implementing own armor sets Minecraft will always search `minecraft:textures/models/armor` for their 3d texture.

Shouldn't Fabric API have an interface for this or at least gain the correct namespace by the item's id?

### PR DESCRIPTION
**Pull request by MrIbby**
Closes #96

Based on #98

### CODE REVIEW

File: fabric-item-materials-v0/src/main/java/net/fabricmc/fabric/mixin/itemmaterial/client/MixinArmorFeatureRenderer.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public class MixinArmorFeatureRenderer {
+	@Shadow
+	@Final
+	private static Map<String, Identifier> ARMOR_TEXTURE_CACHE;
+
+	@Inject(at = @At("HEAD"), method = "method_4174", cancellable = true)
+	private void method_4174(ArmorItem armorItem, boolean isSecond, String suffix, CallbackInfoReturnable<Identifier> info) {
+		Identifier identifier = new Identifier(armorItem.getMaterial().getName());
+		String string_2 = identifier.getNamespace() + ":textures/models/armor/" + identifier.getPath() + "_layer_" + (isSecond ? 2 : 1) + (suffix == null ? "" : "_" + suffix) + ".png";
+		info.setReturnValue(ARMOR_TEXTURE_CACHE.computeIfAbsent(string_2, Identifier::new));
```
Comment (asiekierka) [inline fabric-item-materials-v0/src/main/java/net/fabricmc/fabric/mixin/itemmaterial/client/MixinArmorFeatureRenderer.java]: Does this cancel *always*? For every armor item, even non-Fabric ones? That's horrible...

File: fabric-item-materials-v0/src/main/java/net/fabricmc/fabric/api/itemmaterial/FabricArmorMaterial.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import net.minecraft.item.ArmorMaterial;
+import net.minecraft.item.ArmorMaterials;
+import net.minecraft.recipe.Ingredient;
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.Lazy;
+
+import java.util.function.Supplier;
+
+public class FabricArmorMaterial implements ArmorMaterial {
```
Comment (asiekierka) [inline fabric-item-materials-v0/src/main/java/net/fabricmc/fabric/api/itemmaterial/FabricArmorMaterial.java]: Do we need these in the API? They're nice helpers, but do we need them?

File: fabric-item-materials-v0/src/main/java/net/fabricmc/fabric/mixin/itemmaterial/client/MixinArmorFeatureRenderer.java (hunk 1 above)
Comment (MrIbby) [reply fabric-item-materials-v0/src/main/java/net/fabricmc/fabric/mixin/itemmaterial/client/MixinArmorFeatureRenderer.java]: Yep! What should I do instead?

File: fabric-item-materials-v0/src/main/java/net/fabricmc/fabric/api/itemmaterial/FabricArmorMaterial.java (hunk 2 above)
Comment (MrIbby) [reply fabric-item-materials-v0/src/main/java/net/fabricmc/fabric/api/itemmaterial/FabricArmorMaterial.java]: Nope! 100% not needed.

**Pull request closed without merging**